processor/src/decorators/mod.rs

```diff
@@ -8,10 +8,16 @@ impl Process {
     pub(super) fn execute_decorator(
         &mut self,
         decorator: &Decorator,
+        in_debug_mode: bool,
     ) -> Result<(), ExecutionError> {
         match decorator {
             Decorator::Advice(injector) => self.dec_advice(injector)?,
-            Decorator::AsmOp(_) => (),
+            Decorator::AsmOp(asmop_info) => {
+                if in_debug_mode {
+                    self.decoder
+                        .append_asmop(self.decoder.ops_len(), asmop_info.clone());
```

Comment:
It seems like you are using `self.decoder.ops_len()` as a way to get the current clock cycle. If so, it might be better to use `self.system.clk()`.

Reply:
Thanks. Fixed.